### PR TITLE
Only have one Santa Tracker variant

### PR DESCRIPTION
**Pull request by lptr**
Fixes https://github.com/gradle/gradle-private/issues/3282

We used to have a Java-only variant of Santa Tracker being tested. Now we only test the main variant that includes Kotlin code, too. This includes performance and smoke tests.

I've also added cacheability checks for AGP 7.0 using Santa Tracker, and dropped the code related to AGP 3.6 as that is not supported by Gradle 7.0 anymore.

### CODE REVIEW

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy [hunk 1]
```diff
@@ -33,62 +33,51 @@ class AndroidSantaTrackerSmokeTest extends AbstractAndroidSantaTrackerSmokeTest
...
-                            "See https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_5.html#method_workerexecutor_submit_is_deprecated for more details."
-            )
-        } else if (agpVersion.startsWith('4.1')) {
-            expectDeprecationWarnings(result, "The WorkerExecutor.submit() method has been deprecated. " +
-                    "This is scheduled to be removed in Gradle 8.0. Please use the noIsolation(), classLoaderIsolation() or processIsolation() method instead. " +
-                    "See https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_5.html#method_workerexecutor_submit_is_deprecated for more details.")
-        } else {
-            expectNoDeprecationWarnings(result)
-        }
+        expectDeprecationWarnings(result, "The WorkerExecutor.submit() method has been deprecated. " +
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy:49]: Isn't this deprecation warning gone with AGP 7.0?

**Review by wolfs — COMMENTED** [review]
Looks good. I left a comment.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy:49]: Looks like it's still a thing if the test passes, no? Is there something we need to do about this?

Comment (wolfs) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy:49]: We should have an issue for the problem on the AGP side. Is there already an issue?

Comment (wolfs) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerSmokeTest.groovy:49]: Or maybe this is a problem in the Kotlin plugin? Then we also should have an issue on the Kotlin plugin side.

**Pull request merged**